plutus-core/src/Language/PlutusCore/Parser.y

```diff
@@ -190,6 +167,86 @@ getStaticBuiltinName = \case
     "ifThenElse"               -> Just IfThenElse
     _                          -> Nothing
 
+
+-- TODO: somehow allow the type names to be provided from outside
+mkBuiltinType
+    :: DefaultUni <: uni
+    => AlexPosn -> String -> Parse (Type TyName uni AlexPosn)
+mkBuiltinType tyloc tyname = case tyname of
+  "bool"       -> pure $ mkTyBuiltin @Bool       tyloc
+  "bytestring" -> pure $ mkTyBuiltin @ByteString tyloc
+  "char"       -> pure $ mkTyBuiltin @Char       tyloc
+  "integer"    -> pure $ mkTyBuiltin @Integer    tyloc
+  "string"     -> pure $ mkTyBuiltin @String     tyloc
+  "unit"       -> pure $ mkTyBuiltin @()         tyloc
+  _ -> throwError $ UnknownBuiltinType tyloc tyname
+
+mkBuiltinConstant
+  :: DefaultUni <: uni
+  => AlexPosn -> String -> AlexPosn -> String -> Parse (Term TyName Name uni AlexPosn)
+mkBuiltinConstant tyloc tyname litloc lit  = do
+    val <-
+          case tyname of
+            "bool"       -> case lit of
+                              "true"  -> pure $ Just $ mkConstant litloc True
+                              "false" -> pure $ Just $ mkConstant litloc False
+                              _       -> pure Nothing
+            "bytestring" -> parseByteString tyname litloc lit
+            "char"       -> readConstant @ Char    litloc lit 
+            "integer"    -> readConstant @ Integer litloc lit 
```

Comment:
We should have a single function returning the tag of a type in the universe and then define `mkBuiltinType` and `mkBuiltinConstant` in terms of it, so that we don't need to replicate the dispatching logic.

Reply:
I'm not sure if we can do that yet.  I got some way with doing `mkBuiltinType` like that, but got stuck on, `mkBuiltinConstant` because I really do have to dispatch on the type there: you have to call different functions for different types.  It might be possible to do that once we have a `Parsable` class, but I really don't want to do that in this PR.  I was trying to break this stuff into small subtasks that I could get out of the way quickly without getting stuck in review for ages, so I'd rather merge it as is and do this other stuff later.  Maybe there is some way to do this quickly here, but I spent some time on that and just got stuck.

Reply:
[Time passes ...]  I added a `Parsable` class and eventually got `mkBuiltinConstant` to work generically. I'll tidy that up and commit it.